.github/actions/restore_or_install_byond/action.yml

```diff
@@ -0,0 +1,41 @@
+# This action attempts to restore BYOND from a cache, or to install it otherwise.
+name: Restore or Install BYOND
+description: Attempts to restore a specified BYOND version from cache; if it can't, it installs it.
+
+inputs:
+  release:
+    description: "The release channel of BYOND to install. Either STABLE or BETA."
+    required: false
+    type: string
+    default: "STABLE"
+
+runs:
+  using: composite
+  steps:
+    - name: Configure BYOND version from _build_dependencies.sh
+      shell: bash
+      run: |
+        source _build_dependencies.sh
+        echo "BYOND_MAJOR=$${{ inputs.release }}_BYOND_MAJOR" >> $GITHUB_ENV
+        echo "BYOND_MINOR=$${{ inputs.release }}_BYOND_MINOR" >> $GITHUB_ENV
+
+    # The use of `actions/cache/restore` and `actions/cache/save` here is deliberate, as we want to
+    # save the BYOND install to a cache as early as possible. If we used just `actions/cache`, it
+    # would only attempt to save the cache at the end of a job. This ensures that if a workflow run
+    # is cancelled, we already have a cache to restore from.
+    - name: Restore BYOND cache
+      id: restore_byond_cache
+      uses: actions/cache/restore@v4
+      with:
+        path: ~/BYOND
+        key: ${{ runner.os }}-byond-${{ env.BYOND_MAJOR }}-${{ env.BYOND_MINOR }}
+    - name: Install BYOND
+      if: ${{ !steps.restore_byond_cache.outputs.cache-hit }}
+      shell: bash
+      run: bash tools/ci/install_byond.sh
```

Comment:
The fact this doesn't need the arg specified for the script to pick it up in `$1` scares me.

Reply:
if i'm understanding your concern correctly: _build_dependencies.sh first checks for $BYOND_MAJOR in the environment, before it checks for $1. since we've set that in the step in L19-20 this can move forward without having to specify BETA or STABLE in the first arg.